core/txpool/txpool.go

```diff
@@ -598,93 +579,49 @@ func (pool *TxPool) local() map[common.Address]types.Transactions {
 // This check is meant as an early check which only needs to be performed once,
 // and does not require the pool mutex to be held.
 func (pool *TxPool) validateTxBasics(tx *types.Transaction, local bool) error {
-	// Accept only legacy transactions until EIP-2718/2930 activates.
-	if !pool.eip2718.Load() && tx.Type() != types.LegacyTxType {
-		return core.ErrTxTypeNotSupported
-	}
-	// Reject dynamic fee transactions until EIP-1559 activates.
-	if !pool.eip1559.Load() && tx.Type() == types.DynamicFeeTxType {
-		return core.ErrTxTypeNotSupported
-	}
-	// Reject blob transactions forever, those will have their own pool.
-	if tx.Type() == types.BlobTxType {
-		return core.ErrTxTypeNotSupported
-	}
-	// Reject transactions over defined size to prevent DOS attacks
-	if tx.Size() > txMaxSize {
-		return ErrOversizedData
-	}
-	// Check whether the init code size has been exceeded.
-	if pool.shanghai.Load() && tx.To() == nil && len(tx.Data()) > params.MaxInitCodeSize {
-		return fmt.Errorf("%w: code size %v limit %v", core.ErrMaxInitCodeSizeExceeded, len(tx.Data()), params.MaxInitCodeSize)
-	}
-	// Transactions can't be negative. This may never happen using RLP decoded
-	// transactions but may occur if you create a transaction using the RPC.
-	if tx.Value().Sign() < 0 {
-		return ErrNegativeValue
-	}
-	// Ensure the transaction doesn't exceed the current block limit gas.
-	if pool.currentMaxGas.Load() < tx.Gas() {
-		return ErrGasLimit
-	}
-	// Sanity check for extremely large numbers
-	if tx.GasFeeCap().BitLen() > 256 {
-		return core.ErrFeeCapVeryHigh
-	}
-	if tx.GasTipCap().BitLen() > 256 {
-		return core.ErrTipVeryHigh
+	opts := &ValidationOptions{
+		Config: pool.chainconfig,
+		Accept: map[uint8]struct{}{
+			types.LegacyTxType:     {},
+			types.AccessListTxType: {},
+			types.DynamicFeeTxType: {},
+		},
```

Comment:
Some would say that using a map here is overkill, you could use a single byte and use bitflags. E.g. something like this: 

` Accept : (1 << types.LegacyTxType) || ( 1 << types.AccessListTxType)  ||  ( 1 << types.DynamicFeeTxType) `

Reply:
Fair enough, I'll update it